File: api/submit-score.py
import urllib.request, json, os
import datetime

# Read from env
APP_ID = os.environ.get("FEISHU_APP_ID", "")
APP_SECRET = os.environ.get("FEISHU_APP_SECRET", "")
APP_TOKEN = os.environ.get("FEISHU_APP_TOKEN", "AH5ewoyPaitFb1kHecOcNiN2nL0")
TABLE_ID = os.environ.get("FEISHU_TABLE_ID", "tblEO7PbkyjcWy3l")

def get_tenant_token():
    body = json.dumps({"app_id": APP_ID, "app_secret": APP_SECRET}).encode()
    req = urllib.request.Request(
        "https://open.feishu.cn/open-apis/auth/v3/tenant_access_token/internal",
        data=body, headers={"Content-Type": "application/json"}
    )
    res = json.loads(urllib.request.urlopen(req).read())
    if res.get("code") != 0:
        raise Exception(f"token error: {res.get('msg')}")
    return res["tenant_access_token"]

def write_record(token, data):
    headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json"}
    record = {"fields": data}
    body = json.dumps(record, ensure_ascii=True).encode()
    url = f"https://open.feishu.cn/open-apis/bitable/v1/apps/{APP_TOKEN}/tables/{TABLE_ID}/records"
    req = urllib.request.Request(url, data=body, headers=headers, method="POST")
    res = json.loads(urllib.request.urlopen(req).read())
    return res

def handler(request):
    try:
        method = request.get("httpMethod", "GET")
        if method != "POST":
            return {"statusCode": 405, "body": json.dumps({"error": "Method not allowed"})}
        
        body_raw = request.get("body", "{}")
        payload = json.loads(body_raw)
        
        # Required fields
        required = ["name", "department", "totalScore", "passed"]
        for r in required:
            if r not in payload or payload[r] is None:
                return {"statusCode": 400, "body": json.dumps({"error": f"Missing field: {r}"})}
        
        # Map to Feishu fields
        today_ms = int(datetime.datetime.now().timestamp() * 1000)
        feishu_fields = {
            "主播姓名": payload.get("name", ""),
            "部门": payload.get("department", ""),
            "日期": today_ms,
            "得分": payload.get("totalScore", 0),
            "是否及格": "是" if payload.get("passed") else "否",
            "AI评语": payload.get("feedback", ""),
            "答题用时": payload.get("duration", 0),
            "切屏次数": payload.get("switches", 0)
        }
        
        token = get_tenant_token()
        result = write_record(token, feishu_fields)
        
        if result.get("code") == 0:
            return {
                "statusCode": 200,
                "body": json.dumps({
                    "success": True,
                    "record_id": result["data"]["record"]["record_id"]
                })
            }
        else:
            return {
                "statusCode": 500,
                "body": json.dumps({"error": f"Feishu error: {result.get('msg')}"})
            }
    except Exception as e:
        return {
            "statusCode": 500,
            "body": json.dumps({"error": str(e)})
        }

File: api/submit-score.js
const https = require('https');

// 飞书凭证从环境变量读取（Vercel后台配置），前端不感知
const FEISHU_APP_ID = process.env.FEISHU_APP_ID || '';
const FEISHU_SECRET = process.env.FEISHU_APP_SECRET || '';
const APP_TOKEN = process.env.FEISHU_APP_TOKEN || 'AH5ewoyPaitFb1kHecOcNiN2nL0';
const TABLE_ID = process.env.FEISHU_TABLE_ID || 'tblEO7PbkyjcWy3l';

function feishuFetch(url, options) {
  return new Promise((resolve, reject) => {
    const req = https.request(url, options, (res) => {
      let data = '';
      res.on('data', c => data += c);
      res.on('end', () => {
        try { resolve(JSON.parse(data)); }
        catch(e) { reject(new Error('parse: ' + data.substring(0,200))); }
      });
    });
    req.on('error', reject);
    if (options.body) req.write(options.body);
    req.end();
  });
}

module.exports = async (req, res) => {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
  if (req.method === 'OPTIONS') return res.status(200).end();
  if (req.method !== 'POST') return res.status(405).json({ error: 'Method not allowed' });

  if (!FEISHU_APP_ID || !FEISHU_SECRET) {
    return res.json({ code: -2, msg: 'Server not configured - contact admin' });
  }

  try {
    const payload = req.body || {};

    // 1) 获取飞书token
    const tr = await feishuFetch(
      'https://open.feishu.cn/open-apis/auth/v3/tenant_access_token/internal',
      { method:'POST', headers:{'Content-Type':'application/json'},
        body: JSON.stringify({ app_id: FEISHU_APP_ID, app_secret: FEISHU_SECRET })}
    );
    if (tr.code !== 0) throw new Error('token: ' + tr.msg);
    const token = tr.tenant_access_token;

    // 2) 写入飞书
    const fields = {
      '主播姓名': payload.name || '',
      '部门': payload.department || '',
      '日期': Date.now(),
      '得分': payload.totalScore || 0,
      '是否及格': payload.passed ? '是' : '否',
      'AI评语': payload.feedback || '',
      '答题用时': payload.duration || 0,
      '切屏次数': payload.switches || 0
    };

    const wr = await feishuFetch(
      `https://open.feishu.cn/open-apis/bitable/v1/apps/${APP_TOKEN}/tables/${TABLE_ID}/records`,
      { method:'POST',
        headers:{'Authorization':'Bearer '+token,'Content-Type':'application/json'},
        body: JSON.stringify({fields})}
    );

    if (wr.code === 0) {
      return res.json({ code: 0, success: true, record_id: wr.data.record.record_id });
    } else {
      return res.json({ code: -1, msg: wr.msg || 'write failed' });
    }
  } catch(e) {
    return res.json({ code: -1, msg: e.message });
  }
};
